spec/element_base_spec.rb

```diff
@@ -0,0 +1,94 @@
+require 'spec_helper'
+
+RSpec.describe Kitchen::ElementBase do
+
+  let(:element) do
+    book_containing(html:
+      one_chapter_with_one_page_containing(
+        <<~HTML
+          <div data-type="example" class="class1" id="div1">
+            <p>This is a paragraph.</p>
+          </div>
+        HTML
+      )
+    )
+  end
+
+  describe '#has_class?' do
+    it 'returns true if element has given class' do
+      div_element = element.search('div[data-type="example"]').first
+      expect(div_element.has_class?('class1')).to eq true
+    end
+
+    it 'returns false if element does not have given class' do
+      div_element = element.search('div[data-type="example"]').first
+      expect(div_element.has_class?('class2')).to eq false
+    end
+  end
+
+  describe '#id' do
+    it 'returns the element\'s ID' do
+      div_element = element.search('div[data-type="example"]').first
```

Comment:
instead of repeating this line a few times, do this at the top after `let(:element)`

```ruby
let(:example) { element.first!('div[data-type="example"]') }
```